src/lib.rs

```diff
@@ -59,19 +59,43 @@ impl Shaders {
     }
 }
 
-/// A container for a bindable texture to be used internally.
+/// A container for a bindable texture.
 pub struct Texture {
+    texture: wgpu::Texture,
     bind_group: BindGroup,
+    width: u32,
+    height: u32,
 }
 
 impl Texture {
-    /// Creates a new imgui texture from a wgpu texture.
+    /// Creates a new GPU texture width the specified `width` and `height`.
+    ///
+    /// - `width`: The width of the new texture in pixels.  
+    /// - `height`: The height of the new texture in pixels.  
+    /// - `label`: Identifies the texture in a debugger.
     pub fn new(
-        texture: wgpu::Texture,
-        layout: &BindGroupLayout,
         device: &Device,
+        renderer: &Renderer,
```

Comment:
I think this should be a `TextureConfig` too :)

Reply:
I think we can add many more parameters that the user can change with a `TextureConfig` struct, but I'm not sure which parameters exactly the user should be able to modify. So I'll just add this configuration struct with the current parameters.